packages/aws-cdk-lib/aws-s3-deployment/lib/source.ts

```diff
@@ -262,5 +258,33 @@ export class Source {
     };
   }
 
+  /**
+   * Process objects such that it escapes token output suitable for JSON output.
+   *
+   * @param scope Parent construct scope
+   * @returns Object with with tokens escaped for JSON output.
+   */
+  private static escapeTokens(scope: Construct, obj: any): any {
+    if (Token.isUnresolved(obj)) {
+      // Return tokens as numbers. This is a hack to prevent the JSON serializer to wrap this token as a string.
+      // Stack.toJsonString should take care of escaping the object output for JSON.
+      return Token.asNumber(Stack.of(scope).toJsonString(obj));
```

Comment:
Would be great to see what the string and number representations of the token look like. Maybe add a comment with an example

Reply:
Not sure how we could make this clearer. Number tokens are just random numbers recorded by the CDK token system and token strings are in the form of `"${TOKEN[..]}"`. There's also list tokens in the form of `["#${TOKEN[...]"]`. This info wouldn't add much for contributors who aren't familiar with the token system than what is already written in https://docs.aws.amazon.com/cdk/v2/guide/tokens.html